dateparser/date.py

```diff
@@ -314,16 +309,11 @@ def __init__(self, languages=None, locales=None, region=None, try_previous_local
             raise TypeError("try_previous_locales argument must be a boolean (%r given)"
                             % type(try_previous_locales))
 
-        if not isinstance(use_given_order, bool):
-            raise TypeError("use_given_order argument must be a boolean (%r given)"
-                            % type(use_given_order))
-
-        if not locales and use_given_order:
-            raise ValueError("locales must be given if use_given_order is True")
+        if not (locales or languages) and settings.USE_GIVEN_LANGUAGE_ORDER:
+            raise ValueError("locales or languages must be given if USE_GIVEN_LANGUAGE_ORDER is True")
```

Comment:
Not sure of this failure. If the given list of languages is empty in this case then it should use the defaults. 

I'm thinking on the case where this is combined with a language detection model. For some cases, the prediction could fail and not return any language, but still, dateparser must try its best.

Reply:
I do see your point, and I agree. The question is if this is what's expected: to restrict to the given locales (or languages) or to use those as a starting point but try the other locales anyways.
If the second is true, then removing this check should be the right choice.

Reply:
I can talk about the use case I have in mind: using a page language detector to provide hints to have better date parsing. In this case, what you have is just `hints`:

- In some case, you don't get any language from the language detector. Still, you want dateparser to try its best. 
- In some cases, the list won't be complete. You don't want to remove locales just because the language detector didn't detect them. 

So from my point of view, this list should be just hints: they can change the order of the default locales, but never restrict them.

Reply:
Hi @ivanprado, the `languages` and `locales` parameters have been used historically for restricting the used languages. This is sometimes really important and helps to improve the performance, so I think we can't change them to work as you mentioned.

However, what you say sounds like adding something new like a `exclude_languages` parameter, so you could do something like this:

```python3
date = dateparser.parse("<date string>", languages=['ru', 'tr', 'es'])
if not date:
    date = dateparser.parse("<date string>", exclude_languages=['ru', 'tr', 'es'])
```

What do you think? Could this work for you?

Reply:
Also, on the argument of not raising an exception because the calling code may provide an empty list, the calling code should also be able not to set that setting when that happens. I fail to see the point of removing the exception.

Reply:
@noviluni I see your point. It was not clear to me that `locales` and `languages` were used as contraints to improve performance. The documentation (https://github.com/scrapinghub/dateparser/blob/master/dateparser/__init__.py#L23-L31) is ambiguous, and my first thought was that they would be used as "hints" . I think it would be great to update the documentation to make it clear that they restrict the languages used. 

Given that `languages` and `locales` are for restricting I agree with you than then maybe is not the best place for the use case I'm proposing. What I'm proposing is to be able to propose "hints" for the locales/languages. I don't think `exclude_languages` should be way. 

What do you think about a new parameter `language_hints`? It would be used just to alter the languages/locales priorities.